hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/TestUpdateTable.scala

```diff
@@ -48,235 +48,14 @@ class TestUpdateTable extends HoodieSparkSqlTestBase {
-        spark.sql(s"update $ptTableName set price = price * 1.1, ts = ts + 1 where name = 'a2'")
-        checkAnswer(s"select id, name, price, ts, pt from $ptTableName")(
-          Seq(1, "a1", 10.0, 1000, "2021"),
-          Seq(2, "a2", 22.0, 1001, "2021"),
-          Seq(3, "a2", 33.0, 1001, "2022")
-        )
-
-        spark.sql(s"update $ptTableName set price = price + 5, ts = ts + 1 where pt = '2021'")
-        checkAnswer(s"select id, name, price, ts, pt from $ptTableName")(
-          Seq(1, "a1", 15.0, 1001, "2021"),
-          Seq(2, "a2", 27.0, 1002, "2021"),
-          Seq(3, "a2", 33.0, 1001, "2022")
-        )
-      }
-    })
-  }
-
-  test("Test ignoring case for Update Table") {
-    withRecordType()(withTempDir { tmp =>
-      Seq("cow", "mor").foreach {tableType =>
-        val tableName = generateTableName
-        // create table
-        spark.sql(
-          s"""
-             |create table $tableName (
-             |  ID int,
-             |  NAME string,
-             |  PRICE double,
-             |  TS long
-             |) using hudi
-             | location '${tmp.getCanonicalPath}/$tableName'
-             | options (
-             |  type = '$tableType',
-             |  primaryKey = 'ID',
-             |  preCombineField = 'TS'
-             | )
-       """.stripMargin)
-        // insert data to table
-        spark.sql(s"insert into $tableName select 1, 'a1', 10, 1000")
-        checkAnswer(s"select id, name, price, ts from $tableName")(
-          Seq(1, "a1", 10.0, 1000)
-        )
-
-        // update data
-        spark.sql(s"update $tableName set PRICE = 20 where ID = 1")
-        checkAnswer(s"select id, name, price, ts from $tableName")(
-          Seq(1, "a1", 20.0, 1000)
-        )
-
-        // update data
-        spark.sql(s"update $tableName set PRICE = PRICE * 2 where ID = 1")
-        checkAnswer(s"select id, name, price, ts from $tableName")(
-          Seq(1, "a1", 40.0, 1000)
-        )
-      }
-    })
-  }
-
-  test("Test decimal type") {
-    withTempDir { tmp =>
-      val tableName = generateTableName
-      // create table
-      spark.sql(
-        s"""
-           |create table $tableName (
-           |  id int,
-           |  name string,
-           |  price double,
-           |  ts long,
-           |  ff decimal(38, 10)
-           |) using hudi
-           | location '${tmp.getCanonicalPath}/$tableName'
-           | tblproperties (
-           |  type = 'mor',
-           |  primaryKey = 'id',
-           |  preCombineField = 'ts'
-           | )
-     """.stripMargin)
-
-      // insert data to table
-      spark.sql(s"insert into $tableName select 1, 'a1', 10, 1000, 10.0")
-      checkAnswer(s"select id, name, price, ts from $tableName")(
-        Seq(1, "a1", 10.0, 1000)
-      )
-
-      spark.sql(s"update $tableName set price = 22 where id = 1")
-      checkAnswer(s"select id, name, price, ts from $tableName")(
-        Seq(1, "a1", 22.0, 1000)
-      )
-    }
-  }
```

Comment:
also, can you parametrize some of the update and delete tests for both cases. Just incase some code changes break the unoptimized flow, our tests should catch them. 
Do not make every test parametrized. But a subset of them

Reply:
Parameterized `Test Update Table` in `TestUpdateTable` and `Test Delete Table` in `TestDeleteTable` class.